src/providers/mbed_provider/mod.rs

```diff
@@ -104,7 +97,9 @@ impl MbedProvider {
                         let pc_key_id = key::Id::from_persistent_key_id(key_id);
                         match key::Attributes::from_key_id(pc_key_id) {
                             Ok(_) => {
-                                let _ = local_ids_handle.insert(key_id);
+                                if key_id > mbed_provider.id_counter.load(Relaxed) {
+                                    mbed_provider.id_counter.store(key_id, Relaxed);
+                                }
```

Comment:
It might be more efficient to find the maximum locally, in a `psa_key_id_t` variable and then store it in the `id_counter` at the end. I believe the atomic operations impose restrictions on compiler optimizations.

Reply:
Ah, didn't realise there would be much overhead in just using the atomic variable. I will change it.